src/ORM/Query.php

```diff
@@ -920,6 +925,26 @@ protected function _addDefaultFields()
     }
 
     /**
+     * Sets the default types for converting the fields in the select clause
+     *
+     * @return void
+     */
+    protected function _addDefaultSelectTypes()
+    {
+        $typeMap = $this->typeMap()->defaults();
+        $selectTypeMap = $this->selectTypeMap();
```

Comment:
Why do the 2 type maps need be kept separate? For a select query, could we use only 1 TypeMap?